nofos/guides/templates/guides/guide_edit.html

```diff
@@ -293,38 +289,62 @@ <h3 class="usa-accordion__heading usa-accordion__compare-heading">
                 console.warn(data.message);
             }
 
-            // Handle failed subsection IDs by toggling their checkboxes to opposite value
+            // Handle failed subsection IDs by reverting their checkboxes to previous state
             if (data.failed_subsection_ids && data.failed_subsection_ids.length > 0) {
                 data.failed_subsection_ids.forEach(failedId => {
                     const checkbox = document.querySelector(`input[name="compare_subsections"][value="${failedId}"]`);
                     if (checkbox) {
-                        checkbox.checked = !checkbox.checked;
+                        // Find the checkbox in our value map and revert to current (previous) state
+                        for (let [mapCheckbox, values] of checkboxValueMap.entries()) {
+                            if (mapCheckbox.value === failedId) {
+                                checkbox.checked = values.current;
+                                break;
+                            }
+                        }
                         indicateCheckboxError(checkbox);
                     }
                 });
-
-                showFailureAlert(data.failed_subsection_ids);
-
-                // Update dependent checkboxes after toggling failed ones
-                updateCompareSectionCheckboxes();
-                updateCompareAllSectionsCheckbox();
+            }
+            if (data.status !== 'success') {
+                showMessage(data);
             }
         })
         .catch(error => {
             console.error('Error saving subsection selections:', error);
-        });
-
-        allCheckboxes.forEach(checkbox => {
-            checkbox.disabled = false;
+        })
+        .finally(() => {
+            updateCompareSectionCheckboxes();
+            updateCompareAllSectionsCheckbox();
+            allCheckboxes.forEach(checkbox => {
+                checkbox.disabled = false;
+            });
         });
     }
 
-    function showFailureAlert(failedIds) {
-      // TODO: show a failure alert
+    function showMessage(data) {
+            // Create a simple message div
+            const messageDiv = document.createElement('div');
+            messageDiv.className = `usa-alert usa-alert--${data.status === 'success' ? 'success' : data.status === 'fail' ? 'error' : 'warning'} usa-alert--slim margin-top-2`;
+            messageDiv.innerHTML = `
+                <div class="usa-alert__body">
+                    <p class="usa-alert__text">${data.message}</p>
+                </div>
+            `;
```

Comment:
Let's focus this div if it shows up. I think we may need a "tabindex" to do that.